cipher/cipher_test.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cipher_test
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/apid/apid-core/cipher"
+	"math"
+	"testing"
+)
+
+func TestEvents(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Cipher Suite")
+}
+
+var _ = Describe("APID Cipher", func() {
+
+	Context("AES", func() {
+
+		Context("AES/ECB/PKCS7Padding Encrypt/Decrypt", func() {
+			keys := [][]byte{
+				// 128-bit
+				{2, 122, 212, 83, 150, 164, 180, 4, 148, 242, 65, 189, 3, 188, 76, 247},
+				// 192-bit
+				{2, 122, 212, 83, 150, 164, 180, 4, 148, 242, 65, 189, 3, 188, 76, 247,
+					2, 122, 212, 83, 150, 164, 180, 4},
+				// 256-bit
+				{2, 122, 212, 83, 150, 164, 180, 4, 148, 242, 65, 189, 3, 188, 76, 247,
+					2, 122, 212, 83, 150, 164, 180, 4, 148, 242, 65, 189, 3, 188, 76, 247},
+			}
+
+			plain := [][]byte{
```

Comment:
Suggestion for any future tests, you can define struct with `input`, `encrypted` and `actual` that way you can put in all related data in one struct

Ref: https://github.com/chebrolus/golearn/blob/master/stringutil/palindrome_test.go